src/lightning/fabric/strategies/fsdp.py

```diff
@@ -243,6 +243,12 @@ def setup_optimizer(self, optimizer: Optimizer) -> Optimizer:
     def module_to_device(self, module: Module) -> None:
         pass
 
+    @contextmanager
+    def module_init_context(self) -> Generator[None, None, None]:
+        device_context = torch.device("meta") if _TORCH_GREATER_EQUAL_2_0 else nullcontext()
```

Comment:
Choosing the meta device by default here seems reasonable to me. We can later also put the logic here for torchdistx deferred init if desired. cc @carmocca

Reply:
If this creates problems, how would you suggest the user to opt out?

Same question for the base strategy implementation

Reply:
The context manager is opt-in, so worst case user can remove it in their code. But in the context of example 3 (see PR description), I was thinking that we could make `init_module(empty_weights=None)` the default, meaning the strategy can decide. True would be to force it and False would be opt-out.

Reply:
I agree, but we need to consider that there might be support for `torch.device("meta")` and `torch.device("fake")` in the future so an empty_weights boolean is not future-proof. We might want a string instead

Reply:
Sure, it is certainly good to discuss what the options are and what the name of the argument would be.

Reply:
Do you want to add this argument here or in a separate PR?